staker-rewards/inspector/scripts/inspect-staker-rewards.js

```diff
@@ -0,0 +1,185 @@
+        const isCacheRefreshEnabled = process.env.CACHE_REFRESH !== "off"
+
+        if (!ethHostname) {
+            console.error(clc.red("Wrong ETH_HOSTNAME value"))
+            return
+        }
+
+        validateIntervalTimestamps(intervalStart, intervalEnd)
+
+        if (isDebugDisabled) {
+            console.debug = function() {}
+        }
+
+        const context = await Context.initialize(ethHostname)
+
+        const { cache } = context
+
+        if (isCacheRefreshEnabled) {
+            console.log("Refreshing keeps cache...")
+            await cache.refresh()
+        }
+
+        // get keeps opened within the given interval
+        const openedKeeps = cache
+            .getKeeps() // get keeps with all statuses
+            .filter(keep =>
+                // get keeps whose creation timestamps are within
+                // the given interval
+                intervalStart <= keep.creationTimestamp &&
+                keep.creationTimestamp <= intervalEnd
+            )
+
+        // from keeps opened within the given interval, get the ones which
+        // have been eventually terminated due to keygen fail
+        const failedOpenedKeeps = openedKeeps
+            .filter(keep =>
+                keep.status.name === KeepStatus.TERMINATED &&
+                keep.status.cause === KeepTerminationCause.KEYGEN_FAIL
+            )
+
+        // get keeps closed or terminated within the given interval
+        // but without the ones terminated due to keygen fail as they
+        // are not relevant for the closed keeps SLA
+        const closedKeeps = cache
+            .getKeeps() // get keeps with all statuses
+            .filter(keep => {
+                // get keeps whose statuses have been changed within the
+                // given interval
+                return intervalStart <= keep.status.timestamp &&
+                    keep.status.timestamp <= intervalEnd
+            })
+            .filter(keep =>
+                // get keeps which are currently in the state `closed` or
+                // `terminated` due to causes other than keygen fail
+                keep.status.name === KeepStatus.CLOSED ||
+                (keep.status.name === KeepStatus.TERMINATED &&
+                    keep.status.cause !== KeepTerminationCause.KEYGEN_FAIL)
+            )
+
+        // from keeps closed or terminated (causes other than keygen fail)
+        // within the given interval, get the ones which have been terminated
+        // due to signature fail
+        const failedClosedKeeps = closedKeeps
+            .filter(keep =>
+                keep.status.name === KeepStatus.TERMINATED &&
+                keep.status.cause === KeepTerminationCause.SIGNATURE_FAIL
+            )
+
+        const allOperators = new Set()
+        cache.getKeeps().forEach(keep => {
+            keep.members.forEach(member => allOperators.add(member))
+        })
+
+        const operatorSummary = []
+
```

Comment:
Another comment I would like to make is that I think we should refactor the code for SLA computation so that it's possible to unit test it. And we should add javascript tests to `staker-rewards`. We are doing a lot of quite complex computations here and we need unit tests for them.

Reply:
I have implemented some tests and added them to the Circle CI config.